File: src/controllers/handlers-options.ts
// Third-Party libraries
import opener from 'opener'

// Constants
import { REPOSITORY } from '@/constants/github'

// Utils
import { writeMessage } from '@/utils/console'
import { getErrorMessage } from '@/utils/errors'
import { generateStylelintConfig } from '@/utils/stylelint'
import { exists } from '@/utils/user-os'

// User-Input
import { addStylelintConfigCleanOrder } from '@/user-input/clean-order'
import { getPackageManger } from '@/user-input/package-managers'
import { usingVSCodeEditor } from '@/user-input/using-vscode'

export const handlerOptionBuild = async () => {
  try {
    const packageJsonPath = `${process.cwd()}/package.json`

    const existPackageJsonInTheCurrentDirectory = await exists(packageJsonPath)

    if (!existPackageJsonInTheCurrentDirectory) {
      writeMessage({
        type: 'error',
        message: getErrorMessage('NotFound')
      })

      process.exit(1)
    }

    const packageManagerToUse = await getPackageManger()
    const usingVSCEditor = await usingVSCodeEditor()
    const useStylelintConfigCleanOrder = await addStylelintConfigCleanOrder()

    await generateStylelintConfig({
      packageManagerToUse,
      usingVSCodeEditor: usingVSCEditor,
      useStylelintConfigCleanOrder
    })
  } catch {
    writeMessage({
      type: 'error',
      message: getErrorMessage('Default')
    })

    process.exit(1)
  }
}

export const handlerOptionCollaborate = async () => {
  console.log('Opening the GitHub repository...')

  setTimeout(() => opener(REPOSITORY), 500)
}

File: src/controllers/handlers_options.go
package controllers

import (
	"os"
	"stylelintbc/src/constants"
	userInput "stylelintbc/src/user_input"
	"stylelintbc/src/utils"
)

func HandlerOptionBuild() {
	existPackageJsonInTheCurrentDirectory := utils.Exists("package.json")

	if !existPackageJsonInTheCurrentDirectory {
		utils.WriteMessage(
			utils.WriteMessageProps{
				Type:    "error",
				Message: string(utils.GetErrorMessage("NotFound")),
			},
		)

		os.Exit(1)
	}

	packageManagerToUse := userInput.GetPackageManager()
	addStylelintConfigCleanOrder := userInput.AddStylelintConfigCleanOrder()
	usingVSCodeEditor := userInput.UsingVSCodeEditor()

	utils.GenerateStylelintConfig(utils.GenerateStylelintConfigProps{
		PackageManagerToUse:          packageManagerToUse,
		UseStylelintConfigCleanOrder: addStylelintConfigCleanOrder,
		UsingVSCodeEditor:            usingVSCodeEditor,
	})
}

func HandlerOptionCollaborate() {
	utils.WriteMessage(utils.WriteMessageProps{
		Type:    "info",
		Message: "Opening the GitHub repository...",
	})

	utils.OpenURL(constants.REPOSITORY)
}
